### PR TITLE
build: upgrade

### PR DESCRIPTION
**Pull request by superwf**
* rewrite source by ts.
* compatible with mobx v6.
* use rollup for module and cmd output.
* add module output.
* change declaration file path to module/index.d.ts.
* change type,module,browser,unpkg section in package.json.
* upgrade eslint and babel dependencies.
* update test case.

may be it mobx-react-router should upgrade to v5.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -49,33 +63,48 @@
...
+    "rollup-plugin-terser": "^7.0.2",
     "semantic-release": "^4.3.5",
-    "typescript": "^3.3.4",
+    "ts-node": "^9.0.0",
+    "typescript": "^4.0.5",
     "webpack": "^4.29.6",
     "webpack-cli": "^3.3.0"
   },
   "peerDependencies": {
     "mobx": "^3.0.0 || ^4.0.0 || ^5.0.0",
```
Comment (chensara) [inline package.json:104]: need to add || ^6.0.0?

**Comment by kubk** [comment]
It looks like this library isn't maintained anymore. Would you like to create a separate NPM package?

**Comment by superwf** [comment]
@kubk already published
https://www.npmjs.com/package/@superwf/mobx-react-router

**Comment by VinVinMa** [comment]
I have tried to use this patch with Mobx6, but the UMD build in IE11 gives syntax errors.   You can find the similar error in 

https://github.com/react-dnd/react-dnd/issues/1557

The bundle should be transpiled to IE11 compatible using babel pre-set

See the change in  rollup.config.js 

https://github.com/react-dnd/react-dnd/pull/1621/commits/242a6e177405ed046ce77575921bbc001825156a

**Comment by eek** [comment]
@jimbofreedman @eduardofcgo @alisd23 can you guys check this / merge so we have MobX v6 support?

**Comment by kubk** [comment]
@eek You can use the aforementioned fork for now. The original author of this library doesn't respond to emails.

**Comment by eek** [comment]
@kubk I am using the other repo at the moment, but in the other issue - https://github.com/alisd23/mobx-react-router/issues/74 @alisd23 gave full access to @jimbofreedman and @eduardofcgo to keep the repo updated. Maybe he should have also given permissions to @superwf

**Comment by gcattan** [comment]
@superwf you did such great work with typescript and rollback. Are you still interested in merging some of your changes?

**Pull request closed without merging**